shared/teams/confirm-modals/confirm-remove-from-channel.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Container from '../../util/container'
+import * as Types from '../../constants/types/teams'
+import * as ChatTypes from '../../constants/types/chat2'
+import * as Styles from '../../styles'
+import * as Constants from '../../constants/teams'
+import * as ChatConstants from '../../constants/chat2'
+import * as TeamsGen from '../../actions/teams-gen'
+import {useChannelMeta} from '../common/channel-hooks'
+
+type Props = Container.RouteProps<{
+  members: string[]
+  conversationIDKey: ChatTypes.ConversationIDKey
+  teamID: Types.TeamID
+}>
+
+const ConfirmRemoveFromChannel = (props: Props) => {
+  const members = Container.getRouteProps(props, 'members', [])
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  const conversationIDKey = Container.getRouteProps(
+    props,
+    'conversationIDKey',
+    ChatConstants.noConversationIDKey
+  )
+
+  const waitingKeys = members.map(member => Constants.removeFromChannelWaitingKey(conversationIDKey, member))
+  const waiting = Container.useAnyWaiting(...waitingKeys)
+  const channelMeta = useChannelMeta(teamID, conversationIDKey)
+  const channelname = channelMeta?.channelname ?? ''
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const onCancel = React.useCallback(() => dispatch(nav.safeNavigateUpPayload()), [dispatch, nav])
+
+  // TODO(Y2K-1592): do this in one RPC
+  const onRemove = () => {
+    dispatch(
+      TeamsGen.createChannelSetMemberSelected({
+        clearAll: true,
+        conversationIDKey,
+        selected: false,
+        username: '',
+      })
+    )
+
+    members.forEach(member => console.log(`TODO: ${member} wants to leave channel, not implemented yet`))
+  }
+
+  const wasWaiting = Container.usePrevious(waiting)
+  React.useEffect(() => {
+    if (wasWaiting && !waiting) {
```

Comment:
this action seems like a nice use for `useRPC`, in which case this nav away could be done more explicitly. But maybe a job for the integration ticket.

Reply:
Not sure I follow, we can chat about it in person on the call today.